### PR TITLE
A range of patches to fix warnings in the shared directory

### PR DESCRIPTION
**Pull request by bje-**

### CODE REVIEW

File: shared/lib_util.cpp [hunk 1]
```diff
@@ -326,11 +326,12 @@ std::string util::get_cwd()
...
 #ifdef _WIN32
-	::GetCurrentDirectoryA( 2047, buf );
+	if (::GetCurrentDirectoryA(sizeof(buf), buf) == 0)
+	  return std::string();
 #else
-	::getcwd(buf, 2047);
+	if (::getcwd(buf, sizeof(buf)) == NULL)
+	  return std::string();
 #endif
-	buf[2047] = 0;
```
[the remark this answers is not in the document]
Comment (bje-) [reply shared/lib_util.cpp]: The library functions (getcwd and its Windows equivalent) null terminate the buffer you pass in. There is no need to do it explicitly. Also, the original code needed to know the length of the buffer -- the values 2047 and 2048 appear four times in the original code!  My change uses 2048 once in the declaration and then sizeof(buf) thereafter.

File: shared/lib_weatherfile.cpp [hunk 2]
```diff
@@ -1496,7 +1496,7 @@ bool weatherfile::convert_to_wfcsv( const std::string &input, const std::string
 	else if ( wf.type() == weatherfile::SMW )
 	{
 		fprintf(fp, "Source,Location ID,City,State,Latitude,Longitude,Time Zone,Elevation\n");
-		fprintf(fp, "SMW,%s,%s,%s,%.6lf,%.6lf,%lg,%lg,%d\n", hdr.location.c_str(),
+		fprintf(fp, "SMW,%s,%s,%s.%s,%.6lf,%.6lf,%lg,%lg\n", hdr.location.c_str(),
```
[the remark this answers is not in the document]
Comment (bje-) [reply shared/lib_weatherfile.cpp]: Oh, good catch! Fixed on my branch.

File: shared/lib_powerblock.cpp [hunk 3]
```diff
@@ -1504,7 +1503,6 @@ void CPowerBlock_Type224::ACC(double P_cond_min, int n_pl_inc, double T_ITD_des,
 	T_fan_in_K = T_db + 273.15;									// [K] Fan inlet temperature
 	T_fan_out_K = T_fan_in_K * pow(P_cond_ratio,(R/c_air));
 	T_fan_out = T_fan_out_K - 273.15;							// [C] Fan outlet temperature
-	dT_fan = T_fan_out - T_db;									// [C] Difference in temperature including irreversibilities in fan
```
[the remark this answers is not in the document]
Comment (bje-) [reply shared/lib_powerblock.cpp]: Fixed.

File: shared/lib_powerblock.cpp [hunk 4]
```diff
@@ -1735,7 +1733,6 @@ void CPowerBlock_Type224::HybridHR(/*double fcall,*/ double P_cond_min, int n_pl
 	T_acfan_in_K = T_db + 273.15;  //[K] Fan inlet temperature
 	T_acfan_out_K = T_acfan_in_K * pow(P_cond_ratio,(R/C_air));
 	T_acfan_out = T_acfan_out_K - 273.15;    //[C] Fan outlet temperature
-	dT_acfan = T_acfan_out - T_db;   //[C] Difference in temperature including irreversibilities in fan
```
[the remark this answers is not in the document]
Comment (bje-) [reply shared/lib_powerblock.cpp]: Fixes.

File: shared/lib_powerblock.cpp [hunk 5]
```diff
@@ -1748,7 +1745,6 @@ void CPowerBlock_Type224::HybridHR(/*double fcall,*/ double P_cond_min, int n_pl
 	{
 		//Circulating water pump power
 		h_pcw_in = f_hw_psat(P_amb);     //[J/kg] cw pump inlet enthalpy
-		s_pcw_in = f_s_hw_psat(P_amb);     //[J/kg-K] cw pump inlet entropy
```
[the remark this answers is not in the document]
Comment (bje-) [reply shared/lib_powerblock.cpp]: Fixed.

File: shared/lib_powerblock.cpp [hunk 6]
```diff
@@ -1360,7 +1360,6 @@ void CPowerBlock_Type224::evap_tower(double P_cond_min, int n_pl_inc, double Del
 
 	// Circulating water pump power
 	h_pcw_in = f_hw_psat(P_amb);											// [J/kg] cw pump inlet enthalpy
-	s_pcw_in = f_s_hw_psat(P_amb);											// [J/kg-K] cw pump inlet entropy
```
[the remark this answers is not in the document]
Comment (bje-) [reply shared/lib_powerblock.cpp]: Fixed.

File: shared/lib_powerblock.cpp (hunk 5 above)
[the remark this answers is not in the document]
Comment (bje-) [reply shared/lib_powerblock.cpp]: These should now be good to go.

File: shared/lib_util.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (janinefreeman) [reply shared/lib_util.cpp]: Thanks for the clarifications.

**Comment by janinefreeman** [comment]
Thanks for the changes and the fixes!

**Pull request merged**